src/schema/users.ts

```diff
@@ -2242,6 +2246,24 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         const updatedUser = { ...user, ...data, image: avatar };
         updatedUser.email = updatedUser.email?.toLowerCase();
 
+        if (!updatedUser.acceptedMarketing) {
+          updatedUser.notificationFlags = {
+            ...user.notificationFlags,
+            marketing: {
+              email: NotificationPreferenceStatus.Muted,
+              inApp: NotificationPreferenceStatus.Muted,
+            },
+          };
+        } else {
+          updatedUser.notificationFlags = {
+            ...user.notificationFlags,
+            marketing: {
+              email: NotificationPreferenceStatus.Subscribed,
+              inApp: NotificationPreferenceStatus.Subscribed,
+            },
+          };
+        }
```

Comment:
can you use method similar to updateFlagsStatement to avoid overwriting the whole object?

Reply:
also can we add a test for this?